runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -101,9 +101,12 @@ Java_sun_misc_Unsafe_defineAnonymousClass(JNIEnv *env, jobject receiver, jclass
 	jsize length = env->GetArrayLength(bytecodes);
 
 	if (NULL != vm->sharedClassConfig) {
-		if (J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_SHAREANONYMOUSCLASSES)) {
-			/* if ShareAnonymousClasses is enabled, anonymous classes are shared and for isROMClassShareable() to return true, we need to set the flag of the classLoader */
-			vm->anonClassLoader->flags |= J9CLASSLOADER_SHARED_CLASSES_ENABLED;
+		/* ShareAnonymousClasses is enabled by default so we set the class loader flag for isROMClassShareable() to return true */
+		vm->anonClassLoader->flags |= J9CLASSLOADER_SHARED_CLASSES_ENABLED;
+
+		if (J9_ARE_NO_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_SHAREANONYMOUSCLASSES)) {
+			/* if ShareAnonymousClasses is disabled, anonymous classes are not shared so we unset the flag */
+			vm->anonClassLoader->flags &= (~J9CLASSLOADER_SHARED_CLASSES_ENABLED);
```

Comment:
Can this be done once when creating the `anonClassLoader` rather than on over defineClass?